web3.go

```diff
@@ -268,3 +268,55 @@ func WaitForReceipt(ctx context.Context, client Client, hash common.Hash) (*Rece
 		}
 	}
 }
+
+func FindEventById(abi abi.ABI, id common.Hash) *abi.Event {
+	for _, event := range abi.Events {
+		if event.Id() == id {
+			return &event
+		}
+	}
+	return nil
+}
+func getInputs(args abi.Arguments, indexed bool) []abi.Argument {
+	var out []abi.Argument
+	for _, arg := range args {
+		if arg.Indexed == indexed {
+			out = append(out, arg)
+		}
+	}
+	return out
+}
+
+func ParseReceipt(myabi abi.ABI, receipt *Receipt) (map[string]map[string]interface{}, error) {
```

Comment:
Does this return type assume there is only one of each type of event emitted? Also, is order important?
Seems like we might need something like:
```go
func ParseLogs(abi.ABI, []*types.Logs) ([]Event, error)

type Event struct {
	Name string
	Fields map[string]interface{}
}
```